pkg/svc/extension.go

```diff
@@ -325,34 +333,31 @@ func (s *extensionRegistrySvc) DeleteExtension(ctx context.Context, req *extensi
 // Add a service to an existing extension registered with the FuseML extension
 // registry.
 func (s *extensionRegistrySvc) AddService(ctx context.Context, service *extension.ExtensionService) (res *extension.ExtensionService, err error) {
-	svcRecord, err := s.registry.AddService(ctx, extensionServiceRecordToDomain(service))
+	svc, err := s.registry.AddService(ctx, *service.ExtensionID, extensionServiceToDomain(service))
```

Comment:
what if the `service.ExtensionID` field is nil ? IIRC, there is nothing in the design that enforces that assumption

Reply:
The function that does the conversion to the Domain makes use of the `DerefString` function which in case of `nil` sets it to `""`